cms/search/signal_handlers.py

```diff
@@ -54,26 +58,47 @@ def on_page_deleted(sender: "type[Page]", instance: "Page", **kwargs: Any) -> No
         settings.CMS_SEARCH_NOTIFY_ON_DELETE_OR_UNPUBLISH
         and instance.live
         and instance.specific_class.__name__ not in settings.SEARCH_INDEX_EXCLUDED_PAGE_TYPES
+        and not instance.get_view_restrictions().exists()
     ):
         get_publisher().publish_deleted(instance)
 
 
+@receiver(page_slug_changed)
+def on_page_slug_changed(sender: "type[Page]", instance: "Page", instance_before: "Page", **kwargs: Any) -> None:  # pylint: disable=unused-argument
+    """Called after a Wagtail Page's slug is changed.
```

Comment:
To clarify, it is not triggered when changes are made, but it is still in draft.
```suggestion
    """Called after a slug change to a Wagtail Page is published.
```